src/cli/tools/gleam.ts

```diff
@@ -0,0 +1,80 @@
+import fs from 'node:fs/promises';
+import { join } from 'node:path';
+import { execa } from 'execa';
+import { inject, injectable } from 'inversify';
+import { InstallToolBaseService } from '../install-tool/install-tool-base.service';
+import {
+  CompressionService,
+  EnvService,
+  HttpService,
+  PathService,
+} from '../services';
+
+@injectable()
+export class InstallGleamService extends InstallToolBaseService {
+  readonly name = 'gleam';
+
+  private get ghArch(): string {
+    switch (this.envSvc.arch) {
+      case 'arm64':
+        return 'aarch64';
+      case 'amd64':
+        return 'x86_64';
+    }
+  }
+
+  constructor(
+    @inject(EnvService) envSvc: EnvService,
+    @inject(PathService) pathSvc: PathService,
+    @inject(HttpService) private http: HttpService,
+    @inject(CompressionService) private compress: CompressionService,
+  ) {
+    super(pathSvc, envSvc);
+  }
+
+  override async install(version: string): Promise<void> {
+    /**
+     * @example
+     * @see {@href https://github.com/gleam-lang/gleam/releases/tag/v0.34.1}
+     */
+    const baseUrl = `https://github.com/gleam-lang/gleam/releases/download/v${version}/`;
+
+    const filename = `gleam-v${version}-${this.ghArch}-unknown-linux-musl.tar.gz`;
+    const url = `${baseUrl}${filename}`;
+
+    const checksumFile = await this.http.download({
+      url: `${url}.sha256`,
+    });
+    const expectedChecksum = (await fs.readFile(checksumFile, 'utf-8'))
+      .split('\n')
+      .find((l) => l.includes(filename))
+      ?.split(' ')[0];
+
+    const file = await this.http.download({
+      url,
+      checksumType: 'sha256',
```

Comment:
```suggestion
      checksumType: 'sha512',
```

`checksumType` should be extracted as const and reused on both places